MantidQt/MantidWidgets/src/DataProcessorUI/DataProcessorGenerateNotebook.cpp

```diff
@@ -49,23 +47,22 @@ specified via the corresponding hinting line edit in the view
 @returns ipython notebook string
 */
 DataProcessorGenerateNotebook::DataProcessorGenerateNotebook(
-    std::string name, const std::string instrument,
+    QString name, const QString instrument,
```

Comment:
I agree that it might be going to far in this issue. If it was purely a read of the QString, then `const &` would still be better, since you copy the `cow` handler  itself.